numpy/core/_methods.py

```diff
@@ -55,6 +55,18 @@ def _count_reduce_items(arr, axis):
         items *= arr.shape[ax]
     return items
 
+def _clip(a, a_min=None, a_max=None, out=None):
+    if a_min is None and a_max is None:
+        return um.positive(a, out=out)
+    elif a_min is None:
+        # note: nan-behaviour is incorrect here
+        return um.fmin(a, a_max, out=out)
+    elif a_max is None:
+        # note: nan-behaviour is incorrect here
+        return um.fmax(a, a_min, out=out)
```

Comment:
* max: returns `nan` if either input is NaN
* fmax: returns the non-nan argument if either input is nan
* What we need here: returns the first argument if either input is nan